src/components/ScenarioParameters/components/ScenarioParametersInputs/GenericTable.js

```diff
@@ -209,76 +233,107 @@ export const GenericTable = ({
+    let data;
     try {
-      if (!currentScenarioDatasetList || currentScenarioDatasetList?.length === 0)
-        throw new Error(
-          t(
-            'genericcomponent.table.labels.noDatasetsError',
-            'Impossible to fetch data from dataset because the list of datasets is empty'
-          )
-        );
-      const sourceDatasetId = currentScenarioDatasetList[0];
-      const dynamicValuesConfig = ConfigUtils.getParameterAttribute(parameterData, 'dynamicValues');
-      const query = dynamicValuesConfig?.query;
-      if (!query)
-        throw new Error(
-          t(
-            'genericcomponent.table.labels.noQueryError',
-            'Impossible to fetch data from dataset because there is no twingraph query defined for this parameter. ' +
-              'You can load data manually using Import button'
-          )
-        );
-      const resultKey = dynamicValuesConfig?.resultKey;
-      const { data } = await Api.Datasets.twingraphQuery(organizationId, sourceDatasetId, { query });
-      if (data.length === 0)
-        throw new Error(
-          t(
-            'genericcomponent.table.labels.wrongQueryError',
-            'Returned result is empty, there is probably an error in your query configuration. ' +
-              'Please, check your solution'
-          )
-        );
-      const rowsDict = data.map((row) => row[resultKey]);
-      if (rowsDict.includes(undefined))
-        throw new Error(
-          t(
-            'genericcomponent.table.labels.wrongResultKeyError',
-            `Returned result doesn't have ${resultKey} property.
-      Probably there is an error in resultKey configuration, please, check your solution`,
-            { resultKey }
-          )
-        );
-      const csvRows = AgGridUtils.toCSV(rowsDict, columns);
-      const agGridData = _generateGridDataFromCSV(csvRows, parameterData);
-      if (agGridData.error) {
-        setClientFileDescriptor({
-          tableDataStatus: TABLE_DATA_STATUS.ERROR,
-          errors: agGridData.error,
-        });
-      } else
-        setClientFileDescriptor({
-          name: fileName,
-          file: null,
-          agGridRows: agGridData.rows,
-          errors: null,
-          status: UPLOAD_FILE_STATUS_KEY.READY_TO_UPLOAD,
-          tableDataStatus: TABLE_DATA_STATUS.READY,
-          uploadPreprocess: { content: _uploadPreprocess },
-        });
+      ({ data } = await Api.Datasets.twingraphQuery(organizationId, sourceDatasetId, { query }));
     } catch (error) {
-      const errorComment = error?.response?.status
-        ? t(
-            'genericcomponent.table.labels.notTwingraphDatasetError',
-            'Only twingraph datasets can be used to fetch table data dynamically'
+      setPlaceholderTitle(
+        error?.response?.statusText ?? t('genericcomponent.table.labels.queryFailedError', 'Query failed')
+      );
+      setPlaceholderBody(
+        error?.response?.data?.detail ??
```

Comment:
maybe it's better to put the `error?.response?.data?.detail` in the console and use only custom error message in the placeholder? I'm afraid that sometimes, api can return huge error messages with technical details and it can break the layout